libraries/AP_Motors/AP_MotorsHeli_Single.cpp

```diff
@@ -269,8 +251,24 @@ void AP_MotorsHeli_Single::calculate_scalars()
     // calculate collective mid point as a number from 0 to 1
     _collective_mid_pct = ((float)(_collective_mid-_collective_min))/((float)(_collective_max-_collective_min));
 
-    // calculate factors based on swash type and servo position
-    calculate_roll_pitch_collective_factors();
+    // configure swashplate and update scalars
+    if (_swashplate_type == SWASHPLATE_TYPE_H3) {
+        if (_swash_H3.get_enable() == 0) {
+            _swash_H3.set_enable(1);
+        }
+        _swashplate.set_servo1_pos(_swash_H3.get_servo1_pos());
+        _swashplate.set_servo2_pos(_swash_H3.get_servo2_pos());
+        _swashplate.set_servo3_pos(_swash_H3.get_servo3_pos());
+        _swashplate.set_phase_angle(_swash_H3.get_phase_angle());
+    } else {
+        if (_swash_H3.get_enable() == 1) {
+            _swash_H3.set_enable(0);
+        }
+    }
+    _swashplate.set_swash_type(static_cast<SwashPlateType>((uint8_t)_swashplate_type));
```

Comment:
I guess this is casting _swashplate_type first to a uint8_t and then to a "SwashPlateType".  I suspect the first (uint8_t) case isn't required.